### DISCUSSION TITLE
Building for RKE2, can buildah share storage with local cluster?

### DISCUSSION THREAD

**Discussion by CharlieReitzel** [containers/buildah#3924] [Q&A] [answered]
The title is pretty much my question.  I have a locally running Rancher RKE2 Kubernetes "cluster" running on my local VM.  I use this K8s instance for development testing. 

I'm just starting out with buildah as a way to build images without a Docker daemon running locally.  E.g. I'm still working through how to run buildah as a regular, non-root user.  But that is a different question.

Ideally, I'd like the images buildah builds to show up on the runc runtime RKE2 uses when deploying pods.   Are there appropriate incantations I can use in containers.conf to make this happen?

In a separate-but-related issue, RKE2 can only "see" images in runc in the `k8s.io` namespace.  This may not be the right venue, but is it possible to move an image from one namespace to another or be visible in multiple namespaces?  Put another way, is a runc image namespace a tag that can be added or removed?

Thanks for any pointers.

**Comment by flouthoc** [comment, marked as answer]
Hi @CharlieReitzel , Thanks for creating the issue.

I don't think buildah can directly share storage with your local k8s cluster specially with CRI of RKE2, afaik `cri-o` can still share storage it has a similar format but other CRI managers might not do that (as in your case i think CRI is `containerd`). But I could be wrong so tagging @containers/storage-maintainers for a confirmation.

Easiest way would be to build and push to any local or remote registry.

**Reply by CharlieReitzel** [reply]
Thanks much for the reply.  That tells me I need to go ahead and copy the image to the runtime.

Yes, I can confirm, RKE2 uses containerd.  For example: https://docs.rke2.io/install/containerd_registry_configuration/.

**Reply by rhatdan** [reply]
Yes Buildah shares storage with CRI-O, not containerd.  You could use microshift or CRC.
